lib/block_view.go

```diff
@@ -4160,6 +4169,56 @@ func (bav *UtxoView) ConnectBlock(
 		}
 	}
 
+	// If we're past the PoS cutover, we need to track which validators were active.
+	if blockHeight >= uint64(bav.Params.ForkHeights.ProofOfStake2ConsensusCutoverBlockHeight) {
+		// Get the active validators for the block.
+		var signersList *bitset.Bitset
+		if !desoBlock.Header.ValidatorsVoteQC.isEmpty() {
+			signersList = desoBlock.Header.ValidatorsVoteQC.ValidatorsVoteAggregatedSignature.SignersList
+		} else {
+			signersList = desoBlock.Header.ValidatorsTimeoutAggregateQC.ValidatorsTimeoutAggregatedSignature.SignersList
+		}
+		allSnapshotValidators, err := bav.GetAllSnapshotValidatorSetEntriesByStake()
+		if err != nil {
+			return nil, errors.Wrapf(err, "ConnectBlock: error getting all snapshot validator set entries by stake")
+		}
+		currentEpochNumber, err := bav.GetCurrentEpochNumber()
+		if err != nil {
+			return nil, errors.Wrapf(err, "ConnectBlock: error getting current epoch number")
+		}
+		// TODO: Is it safe to assume that the leader is included in the signers list?
+		// If not, we should explicitly add them here since they were "active".
```

Comment:
Yes! The leader for the previous block will be included in the QC for this block, along with all other validators who voted on the previous block
```suggestion
```